contracts/precompiles/ICS20/ICS20.sol

```diff
@@ -0,0 +1,104 @@
+pragma solidity ^0.8.9;
+
+import "./IICS20Transfer.sol";
+import "./ICS20Bank.sol";
+import "../lib/strings.sol";
+
+abstract contract ICS20 is IICS20Transfer, ICS20Bank
+{
+    mapping(string => address) channelEscrowAddresses;
+
+    constructor(string memory denom_) {
+        _denom = denom_;
+    }
+
+    function sendTx (
+        string calldata denom,
+        uint64 amount,
+        address receiver,
+        string calldata sourcePort,
+        string calldata sourceChannel,
+        uint64 timeoutHeight
+    ) public 
+    {
+        if (!denom.toSlice().startsWith(_makeDenomPrefix(sourcePort, sourceChannel))) 
+        { // sender is source chain
+            require(_transferFrom(_msgSender(), _getEscrowAddress(sourceChannel), denom, amount));
+        } 
+        else 
+        {
+            require(_burn(_msgSender(), denom, amount));
+        }
+
+        emit Transfer(denom, amount, _msgSender(), receiver, sourcePort, sourceChannel, timeoutHeight);
+    }
+
+    function onRecvPacket(Packet.Data calldata packet) external virtual override returns (bytes memory acknowledgement) {
+        FungibleTokenPacketData.Data memory data = FungibleTokenPacketData.decode(packet.data);
+        strings.slice memory denom = data.denom.toSlice();
+        strings.slice memory trimedDenom = data.denom.toSlice().beyond(
+            _makeDenomPrefix(packet.source_port, packet.source_channel)
+        );
+        if (!denom.equals(trimedDenom)) { // receiver is source chain
+            return _newAcknowledgement(
+                _transferFrom(_getEscrowAddress(packet.destination_channel), data.receiver.toAddress(), trimedDenom.toString(), data.amount)
+            );
+        } else {
```

Comment:
you can drop the `else` since the `if` above has a `return`

Reply:
The else allows us to mint `_mint(data.receiver.toAddress(), prefixedDenom, data.amount)` - is that not needed for us?

Reply:
See https://github.com/uber-go/guide/blob/master/style.md#reduce-nesting